src/Language/Docker/Parser/Instruction.hs

```diff
@@ -54,44 +55,79 @@ parseArg = do
       df <- untilEol "the argument value"
       return $ Arg name (Just df)
 
-parseUser :: Parser (Instruction Text)
+parseUser :: (?esc :: Char) => Parser (Instruction Text)
 parseUser = do
   reserved "USER"
   username <- untilEol "the user"
   return $ User username
 
-parseWorkdir :: Parser (Instruction Text)
+parseWorkdir :: (?esc :: Char) => Parser (Instruction Text)
 parseWorkdir = do
   reserved "WORKDIR"
   directory <- untilEol "the workdir path"
   return $ Workdir directory
 
-parseVolume :: Parser (Instruction Text)
+parseVolume :: (?esc :: Char) => Parser (Instruction Text)
 parseVolume = do
   reserved "VOLUME"
   directory <- untilEol "the volume path"
   return $ Volume directory
 
-parseMaintainer :: Parser (Instruction Text)
+parseMaintainer :: (?esc :: Char) => Parser (Instruction Text)
 parseMaintainer = do
   reserved "MAINTAINER"
   name <- untilEol "the maintainer name"
   return $ Maintainer name
 
-parseEntryPoint :: Parser (Instruction Text)
+parseEntryPoint :: (?esc :: Char) => Parser (Instruction Text)
 parseEntryPoint = do
   reserved "ENTRYPOINT"
   Entrypoint <$> arguments
 
-parseOnbuild :: Parser (Instruction Text)
+parseOnbuild :: (?esc :: Char) => Parser (Instruction Text)
 parseOnbuild = do
   reserved "ONBUILD"
   OnBuild <$> parseInstruction
 
-parseComment :: Parser (Instruction Text)
-parseComment = Comment <$> comment
+parsePragma :: (?esc :: Char) => Parser (Instruction Text)
+parsePragma = do
+  void $ char '#'
+  void onlySpaces
+  choice
+    [ parseEscapePragma <?> "an escape",
+      parseSyntaxPragma <?> "a syntax"
+    ]
+
+parseEscapePragma :: Parser (Instruction Text)
+parseEscapePragma = do
+  void $ string "escape"
+  void onlySpaces
+  void $ char '='
+  void onlySpaces
```

Comment:
I'm not 100% happy with this right now. Is there a smarter way to express this?

Reply:
I think a char surrounded by spaces is the `lexeme` or the `symbol` combinator, but I’m ok with some repetition

Reply:
Not quite. Bot do pretty much the same thing, except that symbol is the fixed `string` parser version of lexeme, which accepts a `Parser a`. Both require a whitespace after, which is no use to me because the documentation clearly states that `#escape=\` is valid too.
I've made a custom function lexeme', which works like lexeme but does not require a whitespace after and now I'm happy 😄